### PR TITLE
add priority transition record

### LINKED ISSUE #8677: Record transitions in priority
recording the timestamps and DN of changes in priority would help trace what happened to various requests

### PR DESCRIPTION
**Pull request by ticoann**
fixes #8677 Still need to get the more requirement and test.

### CODE REVIEW

**Comment by ticoann** [comment]
@amaltaro, Alan, please review. Still need to display information and test.

File: src/couchapps/ReqMgr/updates/updaterequest.js [hunk 1]
```diff
@@ -78,9 +89,7 @@ function(doc, req) {
         doc[key] = newValues[key];
 
         // If key is RequestStatus, also update the transition
-        if (key == "RequestStatus") {
-            updateTransition();
-        }
+        updateTransition(key);
```
Comment (amaltaro) [inline src/couchapps/ReqMgr/updates/updaterequest.js]: Directly calling it is certainly cleaner, but there must be a performance penalty for executing L9-13 for every single key. How about we at least define keyAllowed and keyMap outside that function?

File: src/python/WMCore/ReqMgr/Web/ReqMgrService.py [hunk 2]
```diff
@@ -232,6 +232,28 @@ def state_transition(docs):
     out += '</ul></fieldset>'
     return out
 
+def priority_transition(docs):
+    "create html for priority transition format"
+    if  not isinstance(docs, list):
```
Comment (amaltaro) [inline src/python/WMCore/ReqMgr/Web/ReqMgrService.py]: extra space here. Maybe just run the "reformat code" in this module as another commit.

**Review by amaltaro — APPROVED** [review]
Code looks good, I just expressed one opinion about the javascript performance though.

Seangchan, I was going to say we need to update it here as well:
https://github.com/dmwm/WMCore/blob/master/src/python/WMCore/Services/RequestDB/RequestDBWriter.py#L10

but grepping around WMCore, we don't seem to be using it at all. How about you delete it together?

File: src/couchapps/ReqMgr/updates/updaterequest.js (hunk 1 above)
Comment (ticoann) [reply src/couchapps/ReqMgr/updates/updaterequest.js]: I think L9-13 has to run every single key whether it is inside the function or not. The question is how much does it cost to call the function each time. It was true that the cost of calling function is adding additional baggage before. I think these days Javascript engine is much more efficient.  (Not sure with other language, though)
https://www.codereadability.com/performance-cost-javascript-function-call-and-foreach/

But if you think putting the condition check outside is cleaner, we can move that.

File: src/python/WMCore/ReqMgr/Web/ReqMgrService.py (hunk 2 above)
Comment (ticoann) [reply src/python/WMCore/ReqMgr/Web/ReqMgrService.py]: thanks I will fix that.

**Comment by ticoann** [comment]
Alan, Thank you very much for the review. I will run the test and let you know.

**Pull request merged**